rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -163,10 +183,19 @@ NodeGraph::get_node_names() const
       std::string("could not destroy node names: "));
```

Comment:
This exception should only be thrown after trying to `fini` the `node_namespaces_c` struct as well.